pycyphal/transport/serial/_frame.py

```diff
@@ -132,58 +155,73 @@ def parse_from_cobs_image(image: memoryview) -> typing.Optional[SerialFrame]:
         return SerialFrame.parse_from_unescaped_image(memoryview(unescaped_image))
 
     @staticmethod
-    def parse_from_unescaped_image(header_payload_image: memoryview) -> typing.Optional[SerialFrame]:
+    def parse_from_unescaped_image(image: memoryview) -> typing.Optional[SerialFrame]:
         """
         :returns: Frame or None if the image is invalid.
         """
-        if len(header_payload_image) < _HEADER_SIZE:
-            return None
-
-        header = header_payload_image[:_HEADER_SIZE]
-        if not pycyphal.transport.commons.crc.CRC32C.new(header).check_residue():
-            return None
-
-        payload = header_payload_image[_HEADER_SIZE:]
-
-        # noinspection PyTypeChecker
-        (
-            version,
-            int_priority,
-            src_nid,
-            dst_nid,
-            int_data_spec,
-            transfer_id,
-            index_eot,
-        ) = _HEADER_WITHOUT_CRC_FORMAT.unpack_from(header)
-        if version != _VERSION:
+        try:
+            (
+                version,
+                int_priority,
+                source_node_id,
+                destination_node_id,
+                data_specifier_snm,
+                transfer_id,
+                frame_index_eot,
+                user_data,
+                _,  # header CRC
+            ) = _HEADER_FORMAT.unpack_from(image)
```

Comment:
I think you should be using `_HEADER_FORMAT_NO_CRC` here, and `_HEADER_FORMAT` should be removed completely as it is no longer needed. Same goes for the UDP transport.